frontend/src/metabase/containers/AddToDashSelectDashModal.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { Component, ComponentPropsWithoutRef } from "react";
+import { connect } from "react-redux";
+import { t } from "ttag";
+
+import Icon from "metabase/components/Icon";
+import Link from "metabase/core/components/Link";
+import ModalContent from "metabase/components/ModalContent";
+import DashboardPicker from "metabase/containers/DashboardPicker";
+import type { State } from "metabase-types/store";
+import { CreateDashboardFormOwnProps } from "metabase/dashboard/containers/CreateDashboardForm";
+import * as Urls from "metabase/lib/urls";
+import CreateDashboardModal from "metabase/dashboard/containers/CreateDashboardModal";
+import { Card, Dashboard } from "metabase-types/api";
+import { LinkContent } from "./AddToDashSelectDashModal.styled";
+
+function mapStateToProps(state: State) {
+  return {
+    dashboards: state.entities.dashboards,
+  };
+}
+
+interface AddToDashSelectDashModalState {
+  shouldCreateDashboard: boolean;
+}
+
+interface AddToDashSelectDashModalProps {
+  card: Card;
+  onChangeLocation: (location: string) => void;
+  onClose: () => void;
+  dashboards: Record<Dashboard["id"], Dashboard>;
+}
+
+type DashboardPickerProps = ComponentPropsWithoutRef<typeof DashboardPicker>;
+
+class AddToDashSelectDashModal extends Component<
+  AddToDashSelectDashModalProps,
+  AddToDashSelectDashModalState
+> {
+  state = {
+    shouldCreateDashboard: false,
+  };
+
+  navigateToDashboard: Required<CreateDashboardFormOwnProps>["onCreate"] =
+    dashboard => {
+      const { card, onChangeLocation } = this.props;
+
+      onChangeLocation(
+        Urls.dashboard(dashboard, {
+          editMode: true,
+          addCardWithId: card.id,
+        }),
+      );
+    };
+
+  onDashboardSelected: DashboardPickerProps["onChange"] = dashboardId => {
+    const dashboard = this.props.dashboards[dashboardId as Dashboard["id"]];
+    this.navigateToDashboard(dashboard);
+  };
```

Comment:
Should we maybe coerce `dashboardId` to string/number (what we need there)? Like that:

```js
dashboards[String(dashboardId)]
```

Reply:
I'm fine with either approach

Reply:
actually dashboardId can be null|undefined as well, so it's better to have a check if it exists

Reply:
fixed a type, also added additional `if`, so no coercing required